obelisk/cpp/obelisk_cpp/include/obelisk_mujoco_sim_robot.h

```diff
@@ -164,6 +164,62 @@ namespace obelisk {
 
             data_ = mj_makeData(model_);
 
+            // Create the rendering thread
+            std::thread rendering_thread(std::bind(&ObeliskMujocoRobot::SimRender, this));
+
+            mujoco_setup_ = true;
+
+            while (!activation_complete_) {
+                std::this_thread::sleep_for(std::chrono::milliseconds(3));
```

Comment:
is this always consistently enough time? i'd consider bumping up the delay here just to really make sure it's safely compatible with whatever system this is run on that's slower than your machine

Reply:
There is nothing important happening here actually, I am just waiting for the rest of the configuration in other threads to complete, so the timing is not very important.